src/sql/workbench/contrib/notebook/browser/notebookEditor.ts

```diff
@@ -374,15 +371,11 @@ export class NotebookEditor extends EditorPane implements IFindNotebookControlle
 			filters: false
 		};
 		this._notebookModel.cells?.forEach(cell => {
-			this._register(cell.onCellEditModeChanged((isEditMode) => {
-				if (isEditMode) {
-					this._onFindStateChange(changeEvent).catch(onUnexpectedError);
-				}
-			}));
 			this._register(cell.onCurrentEditModeChanged(editMode => {
-				if (editMode !== CellEditModes.WYSIWYG) {
-					this._onFindStateChange(changeEvent).catch(onUnexpectedError);
-				}
+				this._onFindStateChange(changeEvent).catch(onUnexpectedError);
+			}));
+			this._register(cell.onCellPreviewUpdated(previewUpdated => {
```

Comment:
Do one of these end up being fired when cell edit mode is changed as well? (just trying to figure out why the onCellEditModeChanged handler was removed)

Reply:
Yes edit mode sets the mode and in turn fires the onCellEditModeChanged so it find is being called twice. And when we're coming out of edit mode, updatePreview is called which fires the onCellPreviewUpdated that I added.